### ISSUE TITLE
Extension is not working anymore (Firefox 78.0.1 latest stable, on macOS)

### ISSUE THREAD

**Issue by fad1** [mikl/add-to-omnifocus-webextension#2]

**Comment by mikl** [comment]
Hi @fad1, I’ve been using it myself with Firefox 78, as well as the FF 79 beta, and I haven’t experienced any problems. Can you describe what’s happening on your end?

**Comment by fad1** [comment]
Sure, I click the button and nothing happens. It was working prior to upgrading Firefox, a few days ago. Let me know if there's a console or so where I can see what happens when I click the button (error or so). Thanks

PS: I tried on a new profile, after restarting the browser and I get the same issue. I haven't tried restarting the Mac though, let me know if that's useful to try.

**Comment by mikl** [comment]
Sure, if you go to Tools -> Add-ons, and click the wheel-menu, and select the “Debug Add-ons” item, like this:

![Screenshot 2020-07-02 at 18 01 52](https://user-images.githubusercontent.com/10263/86384197-afdb8a80-bc8e-11ea-868e-218f88dff739.png)

Then click “Inspect” on this extension, like this:

![Screenshot 2020-07-02 at 18 02 50](https://user-images.githubusercontent.com/10263/86384496-1a8cc600-bc8f-11ea-9553-68b3b9b56538.png)

That will open the extension’s console in a new tab/window. Unfortunately, that console is a bit of a junk drawer, at least on my machine, there’s a bunch of errors not related to my extension showing up in there.

However, if you watch the console when you click the OmniFocus toolbar button, you should see something like this:

![Screenshot 2020-07-02 at 18 10 55](https://user-images.githubusercontent.com/10263/86384912-b74f6380-bc8f-11ea-88e5-e65358790b2a.png)

In any case, it should tell us something.

Also, bonus question, what happens if you right-click on a link, and use the OmniFocus item in that menu?

**Comment by fad1** [comment]
I tried clicking the button on reddit, no adding window appears, I immediately get in the console: 

"Added task “reddit: the front page of the internet” to OmniFocus. background.js:29:13"

No navigation attempt to the OmniFocus item, and no item created there. (verified in the Changed perspective to make sure nothing got created)

Same behaviour using right click, only difference is line numbers in js file:
Added task “Lockdown has been important for science!” to OmniFocus. background.js:54:11

Also, I have other extensions like tab session manager, uBlock Origin etc and all work properly when I click on their button.

Btw, Firefox stable updated to 78.0.1 not too long ago, and FFox developer to 79. (on your screenshot Ffox dev is at 78) 

edit: (I think that was an old screenshot, just making sure..)
Anything else I can provide?

**Comment by mikl** [comment]
> FFox developer to 79. (on your screenshot Ffox dev is at 78)

Yeah, I get the same results on regular Firefox 78 as well as FDE 79.0b2. It works everywhere.

If you don’t get any errors related to the button click, I don’t really understand what’s going wrong. Have you tried restarting Firefox, OmniFocus, or even macOS? Are you on the macOS 11 beta? Maybe something is blocking the `omnifocus:///` URL, uBlock maybe?

If all that fails, you could also try adding some debugging code to background.js. If you clone this repo to your local machine, you can use the “Install temporary add-on” function in the Add-on Debug screen to install your own copy.

**Comment by fad1** [comment]
Ok, so I got back to the console and I found the following: (I copied everything before clearing everything and retrying to making a screenshot but I would not get anything anymore.)


TypeError: PrecompiledScript.executeInGlobal: Argument 1 is not an object. 2 ExtensionContent.jsm:567:25
Added task “reddit: the front page of the internet” to OmniFocus. background.js:29:13
can't access property "id", port.sender.tab is undefined index.js:1748
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867344 ConduitsChild.jsm:108
Promise rejected after context unloaded: 
promisify.js:12
Added task “TIFU by having the username “Soundman1488” for 15+ years of being on the internet and been unknowingly identifying myself as a Nazi. : tifu” to OmniFocus. 2 background.js:29:13
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867671 ConduitsChild.jsm:108
Error: CallResult for closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867671: ({childId:"{7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867671", callId:412316867674, path:"storage.sync.get", result:{}}) ConduitsChild.jsm:167:13
Promise rejected after context unloaded: 
promisify.js:12
TypeError: PrecompiledScript.executeInGlobal: Argument 1 is not an object. ExtensionContent.jsm:567:25
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867679 ConduitsChild.jsm:108
Promise rejected after context unloaded: 
promisify.js:12
can't access property "id", port.sender.tab is undefined 2 index.js:1748
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867691 ConduitsChild.jsm:108
Promise resolved after context unloaded
promisify.js:12
can't access property "id", port.sender.tab is undefined index.js:1748
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867719 ConduitsChild.jsm:108
can't access property "id", port.sender.tab is undefined index.js:1748
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867730 ConduitsChild.jsm:108
Promise rejected after context unloaded: 
promisify.js:12
Added task “Introduction” to OmniFocus. background.js:29:13
can't access property "id", port.sender.tab is undefined index.js:1748
Promise resolved while context is inactive
promisify.js:12
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867749 ConduitsChild.jsm:108
can't access property "id", port.sender.tab is undefined index.js:1748
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867778 ConduitsChild.jsm:108
can't access property "id", port.sender.tab is undefined index.js:1748
Promise resolved while context is inactive
promisify.js:12
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316867791 ConduitsChild.jsm:108
Added task “Introduction” to OmniFocus. background.js:29:13
TypeError: PrecompiledScript.executeInGlobal: Argument 1 is not an object. 2 ExtensionContent.jsm:567:25
Uncaught (in promise) Error: Invalid tab ID: 29 undefined
Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist.
    blockFacebookSubResources moz-extension://1f70584c-1a49-6843-b377-1002de3586f2/background.js:528
background.js:528:20
Uncaught (in promise) Error: Invalid tab ID: 31 undefined
Uncaught (in promise) Error: Invalid tab ID: 31 undefined
Uncaught (in promise) Error: Invalid tab ID: 33 undefined
Uncaught (in promise) Error: Invalid tab ID: 33 undefined
Uncaught (in promise) Error: Invalid tab ID: 33 undefined
Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist.
    blockFacebookSubResources moz-extension://1f70584c-1a49-6843-b377-1002de3586f2/background.js:528
background.js:528:20
Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist.
    blockFacebookSubResources moz-extension://1f70584c-1a49-6843-b377-1002de3586f2/background.js:528
background.js:528:20
Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist.
    blockFacebookSubResources moz-extension://1f70584c-1a49-6843-b377-1002de3586f2/background.js:528
background.js:528:20
Uncaught (in promise) Error: Invalid tab ID: 35 undefined
Unchecked lastError value: Error: can't access property "browser", tab is null 2 index.js:1779
TypeError: PrecompiledScript.executeInGlobal: Argument 1 is not an object. 2 ExtensionContent.jsm:567:25
can't access property "id", port.sender.tab is undefined index.js:1748
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316870726 ConduitsChild.jsm:108
Promise rejected after context unloaded: 
promisify.js:12
can't access property "id", port.sender.tab is undefined index.js:1748
sendRemoveListener on closed conduit {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}.412316870740 ConduitsChild.jsm:108
Promise rejected after context unloaded: 
promisify.js:12
TypeError: PrecompiledScript.executeInGlobal: Argument 1 is not an object. ExtensionContent.jsm:567:25

​

**Comment by fad1** [comment]
I think I found the solution. 

The problem is that the profile was new. Navigating to omnifocus:/// on a new profile gives this window. Prior to clicking "open link" and checking "Remember my choice for OmniFocus links", the extension does not work, after doing so once, it does.

![image](https://user-images.githubusercontent.com/34130556/86396300-abec4000-bc6f-11ea-92b2-8b7b113cb719.png)

**Comment by mikl** [comment]
A-ha, that would explain things. I’ve long since accepted that on both my browsers, so I never experienced that. I guess I’ll need to find a way to document this, or maybe even a workaround.

**Comment by fad1** [comment]
True, idk if that's a behaviour that changed with this FFox version or present in previous ones as well (where Launch application window does not appear from an extension click).

People who never opened a omnifocus:/// link would have the issue. (new Mac, new FFox install); a quick workaround would probably be the display, at first click, of a window with a link to "omnifocus:///" explaining they have to click it and remember the choice, open link once before.

Great little extension :)
